content/blog/atomic-design-for-spas.md

```diff
@@ -0,0 +1,213 @@
+so I would recommend taking 2 minutes to read it beforehand.
+
+You can find a basic clone of [HackerNews](https://news.ycombinator.com/) on GitHub: <https://github.com/Keats/hn-react> made using
+the principles described in this post.
+
+Something missing from the repository that I use in my own projects is [StoryBook](https://storybook.js.org/) to easily have
+a visual overview of every component - an invaluable tool.
+
+Lastly, while the title doesn't mention a specific framework, all the samples will use React. The
+ideas presented in this post will work equally well in Vue.js or any other component-based
+framework.
+
+
+## Self-contained components
+That is something Vue.js users can get behind!
+Each component has now its own folder containing everything it needs to be used, except
+for other components as we will see in a bit.
+
+What this means is that every component folder looks very similar to something like:
+
+```bash
+input
+├── index.tsx
+└── _style.scss
+```
+
+You can replace `tsx` by `js` if you are not using [TypeScript](https://www.typescriptlang.org/).
+
+As you can see, the folder also contains a `_style.scss`. There might be some use-cases for CSS in JS but I still haven't seen a compelling one
+so I am sticking with [Sass](https://sass-lang.com/). You can use whatever floats your boat though.
+Those files are only targeting specific CSS classes of that component, which solves potentially overlapping styling.
+
+The main Sass file will import each component individual style after setting up some base rules like fonts or vertical rhythm as can be seen in
+[app.scss](https://github.com/Keats/hn-react/blob/91bb067276006b0ea9d191f753112a5688b7c518/src/style/app.scss).
+
+If I am using [GraphQL](https://graphql.org/) with [Apollo](https://www.apollographql.com/), I will also have a `queries.tsx` in there with the queries
+used by the component as well as the types:
+
+
+```tsx
+import gql from "graphql-tag";
+import {Query} from "react-apollo";
+
+import {IOrganization} from "../../../types";
+
+export const ALL_ORGS_QUERY = gql`
+  query allOrgs {
+    organizations {
+      id
+      name
+      slug
+
+      members {
+        id
+        role
+
+        user {
+          id
+          name
+          timezone
+        }
+      }
+    }
+  }
+`;
+
+export interface IData {
+  organizations: Array<IOrganization>;
+}
+
+export class AllOrgsQuery extends Query<IData> {}
+```
+
+You can find more about using TypeScript with Apollo [in their docs](https://www.apollographql.com/docs/react/recipes/static-typing.html).
+
+Now that we have seen how I organise each individual component, let's dive in the Atomic Design part.
+
+## Atomic Design
+
+If you have read the article introducing Atomic Design, you can probably guess that your `components` folder is going to look like that:
+
+```bash
+.
+├── atoms
+├── molecules
+├── organisms
```

Comment:
mouarf. atoms/molecules/organisms are a nice analogy, but from there to name your production code this way ....  xD

Reply:
you get used to it!